src/software/jetson_nano/thunderloop.cpp

```diff
@@ -434,4 +463,14 @@ void Thunderloop::updateErrorCodes()
     {
         robot_status_.mutable_error_code()->Add(TbotsProto::ErrorCode::HIGH_BOARD_TEMP);
     }
+
+    // this is declared as static since we really only want to open this file once and
+    // pull from a buffer. clear would be called to ensure that std::getline would return
+    // the latest changes!
+    static std::ifstream log_file(PATH_TO_RINGBUFFER_LOG);
```

Comment:
this'd be equivalent to declaring this field as a Thunderloop class member. we'd still get the benefits of opening the file once and pulling from it without re-opening multiple file descriptors to the dmesg logs

Reply:
we tend not to use `static` unless for constants.

Reply:
I have redeclared this as a member variable.